iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/tsfile/container/scan/TsFileInsertionScanDataContainer.java

```diff
@@ -332,35 +331,54 @@ private void moveToNextChunkReader() throws IOException, IllegalStateException {
           return;
         case MetaMarker.VALUE_CHUNK_HEADER:
         case MetaMarker.ONLY_ONE_PAGE_VALUE_CHUNK_HEADER:
-          chunkHeader = tsFileSequenceReader.readChunkHeader(marker);
+          if (Objects.isNull(firstChunkHeader4NextSequentialValueChunks)) {
+            chunkHeader = tsFileSequenceReader.readChunkHeader(marker);
 
-          if (Objects.isNull(currentDevice)
-              || !pattern.matchesMeasurement(currentDevice, chunkHeader.getMeasurementID())) {
-            tsFileSequenceReader.position(
-                tsFileSequenceReader.position() + chunkHeader.getDataSize());
-            break;
-          }
+            // Do not record empty chunks
+            if (chunkHeader.getDataSize() == 0) {
+              break;
+            }
```

Comment:
Actually, empty chunks SHOULD also be recorded. Consider the following case:
There are two measurements in an aligned series: s1 and s2, each of them has two chunks.
They are grouped by 1 series 1 group, and then the result will be:
[TimeChunk1] [s1 Chunk1] [TimeChunk2] [s1 Chunk2] [s2 Chunk1] [s2 Chunk2].
Let us assume the [s2 Chunk1] to be empty. If you do not record it, the associated time chunk of [s2 Chunk2] will be marked as [TimeChunk1], which is incorrect.